File: Data Structures & Algorithms/validate-parentheses/submission-3.cpp
class Solution {
public:
   



bool isValid(string s) {
       stack<char> st;
       
       for (auto ch : s) {
       if (ch == '{'){
              st.push('{');
       } else if( ch == '('){
              st.push('(');
       } else if (ch == '['){
              st.push('[');
       }else if (ch == '}'){
              if (!st.empty() and st.top()=='{'){
                     st.pop();
              }else {
                     return false;
              }
       }else if (ch == ']'){
              if (!st.empty() and st.top()=='['){
                     st.pop();
              }else {
                     return false;
              }
       }else if (ch == ')'){

              if (!st.empty() and st.top()=='('){
                     st.pop();
              }else {
                     return false;
              }
       }
       }
       return st.empty();
       

    }

};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-3.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
 	maxDepth(root) {
		if (!root) return 0;
		const stack = [[root, 1]];
		let res = 0;
		while (stack.length) {
			const [top, depth] = stack.pop();
			res = Math.max(res, depth);
			if (top.left)
				stack.push([top.left, depth + 1]);
			if (top.right)
				stack.push([top.right, depth + 1]);
		}
		return res;

	}


}
